File: dotnet-docs-viewer/Common/Error/Response/ErrorResponse.cs
using System.Text.Json.Serialization;

namespace WebApp.Common.Error
{
    public record ErrorResponse
    {
        public string Message { get; init; } = string.Empty;
        public string Code { get; init; } = string.Empty;
        public int Status { get; init; }
        
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public ErrorInfo? Errors { get; init; }
    }
}

File: src/main/java/com/docs/viewer/global/error/dto/response/ErrorResponse.java
package com.docs.viewer.global.error.dto.response;

import com.docs.viewer.global.error.dto.ErrorInfo;
import lombok.Builder;
import lombok.Getter;

@Builder
@Getter
public class ErrorResponse {

    private String message;
    private String code;
    private int status;
    private ErrorInfo errors;
}
